### PR TITLE
Add venv to /bin/sh shells

### PR DESCRIPTION
**Pull request by AndreKurait**
### Description
Add venv to /bin/sh shells

Fixes a pipenv issue where we were not failing when Pipfile did not match Pipfile.lock due to the --ignore-pipfile argument. Now fails with`Your Pipfile.lock (8ca13e) is out of date.  Expected: (e8dad6).`

* Category: Bug Fix
* Why these changes are required? Enable more consistency
* What is the old behavior before changes and new behavior after changes? Jenkins is failing because it uses /bin/sh which does not run .bashrc

### Issues Resolved
N/A

Is this a backport? If so, please add backport PR # and/or commits #

### Testing
Verified with docker exec /bin/bash and /bin/sh

### Check List
- [x] New functionality includes testing
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: TrafficCapture/dockerSolution/README.md [hunk 1]
```diff
@@ -110,3 +110,12 @@ A problem occurred evaluating project ':trafficCaptureProxyServer'.
...
 
 ```
+
+## Development
+
+### Updating Pipfile
+
+When updating Pipfiles, run the following command to update all Pipfile.lock in subdirectories
+```bash
+  find . -type f -name Pipfile -execdir bash -c 'PIPENV_IGNORE_VIRTUALENVS=true pipenv lock --python 3.11' \;
```
Comment (mikaylathompson) [inline TrafficCapture/dockerSolution/README.md:120]: What does this do the user's current environment? Does it install everything in every pipfile? I can't actually tell from the `pipenv lock` documentation.

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/Dockerfile [hunk 2]
```diff
@@ -42,21 +42,23 @@ RUN chmod ug+x /root/loadServicesFromParameterStore.sh
 WORKDIR /root
 COPY lib /root/lib
 WORKDIR /root/lib/console_link
-RUN pipenv install --deploy --ignore-pipfile
+RUN pipenv install --deploy
```
Comment (mikaylathompson) [inline TrafficCapture/dockerSolution/src/main/docker/migrationConsole/Dockerfile:45]: This change is what causes it to throw an error if the lockfile is out of date?

**Review by mikaylathompson — APPROVED** [review]

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/Dockerfile (hunk 2 above)
Comment (AndreKurait) [reply TrafficCapture/dockerSolution/src/main/docker/migrationConsole/Dockerfile:45]: Yes. --deploy and --ignore-pipfile are contradictory and the --ignore-pipfile took precedence.

--deploy is what we want. Install from Pipfile.lock, but fail if it doesn't reflect the Pipfile

File: TrafficCapture/dockerSolution/README.md (hunk 1 above)
Comment (AndreKurait) [reply TrafficCapture/dockerSolution/README.md:120]: It doesn't have any side effects in the user environment. Lock just resolves and updates
```
(python3_venv) ➜  dockerSolution git:(simplifyPipfile) ✗ find . -type f -name Pipfile -execdir bash -c 'PIPENV_IGNORE_VIRTUALENVS=true pipenv lock --python 3.11' \;
Locking [packages] dependencies...
Building requirements...
Resolving dependencies...
✔ Success!
Locking [dev-packages] dependencies...
Updated Pipfile.lock (fff897626f6e22e3ccfe3cbd4392514d3a41679372c7eb1e5c790783485441c5)!
Locking [packages] dependencies...
Building requirements...
Resolving dependencies...
✔ Success!
Locking [dev-packages] dependencies...
Building requirements...
Resolving dependencies...
✔ Success!
Updated Pipfile.lock (914625b4e6d78413f4259907f8b7debe21f1b33fe35e11743809688e335e6d28)!
Locking [packages] dependencies...
Building requirements...
Resolving dependencies...
✔ Success!
Locking [dev-packages] dependencies...
Updated Pipfile.lock (333a2a22976ccc7f9bd36b9d12956616361b20f2f51c538ddeb8bc706f9c9834)!
Locking [packages] dependencies...
Building requirements...
Resolving dependencies...
✔ Success!
Locking [dev-packages] dependencies...
Building requirements...

```

**Pull request merged**